test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -341,21 +355,29 @@ func runFuncWithRecover(f pulumi.RunFunc) pulumi.RunFunc {
 	}
 }
 
-func shouldRetryError(err error) bool {
+type retryType string
+
+const (
+	reUp     retryType = "ReUp"     // Retry the up operation
+	reCreate retryType = "ReCreate" // Retry the up operation after destroying the stack
+	noRetry  retryType = "NoRetry"
+)
+
+func shouldRetryError(err error) retryType {
 	// Add here errors that are known to be flakes and that should be retried
 	if strings.Contains(err.Error(), "i/o timeout") {
-		return true
+		return reCreate
```

Comment:
In which situation does it happen?

Reply:
It happens sometime on the CI, the remote host is successfully created but unreachable. For example https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/432046614

Reply:
Seems like it happens on some specific AMIs, there's perhaps something faulty to investigate.

Reply:
I had a similar flake on `pre_e2e_test` with 

```
reading EC2 Instance (i-014409b598296e1f2): couldn't find resource
```

 https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/441508406, succeeded on retry. Will take a look at other instances of this error. I was discussing possibly to have a list of known errors that we allow to retry on

Reply:
It seems to appear mainly on `Amazon Linux` images, but we also have it sometime on `Centos 6` if I watch the stats for last two weeks. It happens on 4 different AMIs, some owned by us but some owned by Amazon. I am not sure we want to spend a lot of time investigating it since this is impossible to reproduce :/

Reply:
It seems to be a very low flake rate, I agree with not spending too much time investigating, unless there is a clear pattern. We can keep track of those retries:I was going to write "we could send an event to DD", but it's already handled ❤️

Reply:
Surprising to see from upstream AMIs but LGTM